Lib/test/test_os.py

```diff
@@ -4503,6 +4503,23 @@ def test_times(self):
             self.assertEqual(times.elapsed, 0)
 
 
+@requires_os_func('fork')
+@requires_os_func('waitpid')
+class ForkTests(unittest.TestCase):
+    def test_fork(self):
+        # bpo-42540: ensure os.fork() with non-default allocator
+        code = """if 1:
+            import os
+            pid = os.fork()
+            if pid != 0:
+                _, exitcode = os.waitpid(pid, 0)
```

Comment:
Can you use `test.support.wait_process(pid, exitcode=0)`? It prevents to hang if the child proces hangs.

You can remove `@requires_os_func('waitpid')` in that case, os.fork() implies that os.waitpid() exists.